src/cpp/session/modules/SessionCodeTools.R

```diff
@@ -2441,27 +2442,33 @@
    available <- .rs.availablePackages()
    if (is.null(available$value))
       return(character())
-   
+
    # read the associated source file
    contents <- .rs.readSourceDocument(id)
    if (is.null(contents))
       return(character())
 
    # parse to find packages
    packages <- .rs.parsePackageDependencies(contents, extension)
-   
+   # return early if no R code is parsed
+   if (length(packages) == 0)
+      return(character())
+
+   # If file is qmd or rmd, and there is R code, rmarkdown is REQUIRED
+   if (identical(extension, ".Rmd") || identical(extension, ".qmd"))
+      packages <- c("rmarkdown", packages)
```

Comment:
Nit: only add rmarkdown if it's not already a dependency? (For chunks that might be using `library(rmarkdown)` already)

Reply:
I think this is not going to make a difference, because ultimately we're just adding rmarkdown to the list of known dependencies before doing `missing <- setdiff(packages, installed)` at which point it doesn't matter if rmarkdown shows up once or more than once in packages, it will show up 0 or 1 times in missing depending on whether it's installed or not. I think filtering it out before the set different is just a redundant extra step